crates/ra_hir/src/nameres/tests.rs

```diff
@@ -265,6 +265,51 @@ fn glob_across_crates() {
     );
 }
 
+#[test]
+fn edition_2015_imports() {
+    use ra_db::{CrateGraph, Edition};
+    let mut db = MockDatabase::with_files(
+        "
+        //- /main.rs
+        mod foo;
+        mod bar;
+
+        //- /bar.rs
+        struct Bar;
+
+        //- /foo.rs
+        use bar::Bar;
+        use other_crate::FromLib;
+
+        //- /lib.rs
+        struct FromLib;
+    ",
+    );
+    let main_id = db.file_id_of("/main.rs");
+    let lib_id = db.file_id_of("/lib.rs");
+    let foo_id = db.file_id_of("/foo.rs");
+
+    let mut crate_graph = CrateGraph::default();
+    let main_crate = crate_graph.add_crate_root(main_id, Edition::Edition2015);
+    let lib_crate = crate_graph.add_crate_root(lib_id, Edition::Edition2018);
+    crate_graph.add_dep(main_crate, "other_crate".into(), lib_crate).unwrap();
+
+    db.set_crate_graph(Arc::new(crate_graph));
```

Comment:
There's now a more declarative API for setting the crate graph:

https://github.com/rust-analyzer/rust-analyzer/pull/784/files#diff-2e7269539c8928d62c4fb71200a3a2a9R249

We could extend it to support specifying editions as well. 

I am also toying with the idea to make a single method for specifying fixtures, which accepts `ron::Value`.

Reply:
Yeah, I wasn't sure whether it was worth extending the macro for this one test case...

Reply:
I wasn't sure initially, but I imagine that we might face more cases where we care about editions. I think taking an optional edition literal (and defaulting to 2018) shouldn't be too hard.

Reply:
Ok, I extended the macro.